src/container/cli.c

```diff
@@ -871,70 +825,38 @@ dc_cont_open(tse_task_t *task)
 	daos_cont_open_t	*args;
 	struct dc_pool		*pool;
 	struct dc_cont		*cont;
+	const char		*label;
+	uuid_t			 uuid;
 	int			 rc;
 
 	args = dc_task_get_args(task);
 	cont = dc_task_get_priv(task);
 
-	if (uuid_is_null(args->uuid) || args->coh == NULL)
+	if (args->coh == NULL)
 		D_GOTO(err, rc = -DER_INVAL);
 
-	pool = dc_hdl2pool(args->poh);
-	if (pool == NULL)
-		D_GOTO(err, rc = -DER_NO_HDL);
-
-	if (cont == NULL) {
-		cont = dc_cont_alloc(args->uuid);
-		if (cont == NULL)
-			D_GOTO(err_pool, rc = -DER_NOMEM);
-		uuid_generate(cont->dc_cont_hdl);
-		cont->dc_capas = args->flags;
-		dc_task_set_priv(task, cont);
-	}
-
-	D_DEBUG(DF_DSMC, DF_CONT": opening: hdl="DF_UUIDF" flags=%x\n",
-		DP_CONT(pool->dp_pool, args->uuid), DP_UUID(cont->dc_cont_hdl),
-		args->flags);
-
-	rc = dc_cont_open_internal(task, pool);
-	if (rc)
-		goto err_cont;
-
-	return rc;
-
-err_cont:
-	dc_cont_put(cont);
-err_pool:
-	dc_pool_put(pool);
-err:
-	tse_task_complete(task, rc);
-	D_DEBUG(DF_DSMC, "failed to open container: "DF_RC"\n", DP_RC(rc));
-	return rc;
-}
-
-int
-dc_cont_open_lbl(tse_task_t *task)
-{
-	daos_cont_open_t		*args;
-	struct dc_pool			*pool;
-	struct dc_cont			*cont;
-	uuid_t				 null_uuid;
-	int				 rc;
-
-	args = dc_task_get_args(task);
-	cont = dc_task_get_priv(task);
-	/* open bylabel, uuid will be in reply */
-	uuid_clear(null_uuid);
-
-	if ((args->label == NULL) || args->coh == NULL)
+	if (daos_uuid_valid(args->uuid)) {
+		/** Backward compatibility, we are provided a UUID */
+		label = NULL;
+		uuid_copy(uuid, args->uuid);
+	} if (daos_label_is_valid(args->cont)) {
```

Comment:
(style)  trailing statements should be on next line (or did you mean 'else if'?)